ostree/ostree_dest.go

```diff
@@ -430,16 +402,12 @@ func (d *ostreeImageDestination) Commit(ctx context.Context) error {
 		return err
 	}
 
-	var selinuxHnd *C.struct_selabel_handle
-
-	if os.Getuid() == 0 && selinux.GetEnabled() {
-		selinuxHnd, err = C.selabel_open(C.SELABEL_CTX_FILE, nil, 0)
-		if selinuxHnd == nil {
-			return errors.Wrapf(err, "cannot open the SELinux DB")
-		}
-
-		defer C.selabel_close(selinuxHnd)
+	mac := new(mandatoryAccessControl)
+	err = mac.Open()
```

Comment:
It was my first approach, but I probably failed due to lack of Go expertise.

How to return the initialized object if everything is fine, but an error otherwise?

Reply:
Typically, such functions return a pointer, and
```go
if failed {
   return nil, someError
}
return &mandatoryAccessControl{
   // initialized members
}
```

or, perhaps just
```go
if failed {
   return mandatoryAccessControl{}, someError
}
return mandatoryAccessControl{
   // initialized members
}
```

Assuming that the callers do check the error return value, both approaches are equivalent in effect (and the performance impact is not really worth worrying about).

Reply:
I strongly modified the code and this issue should be fixed by now.

I hope I got a more Go idiomatic implementation.